src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,168 @@
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useSearchParamsContext } from '../SearchParamsContext';
 
 export const PeoplePage = () => {
+  const { searchParams } = useSearchParamsContext();
+  const navigate = useNavigate();
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  const [sortedPeople, setSortedPeople] = useState<Person[]>([]);
+
+  const query = searchParams.get('query') || '';
+  const selectedCenturies = searchParams.getAll('centuries');
+  const selectedSex = searchParams.get('sex');
+
+  const filteredByQuery = people.filter(person => {
+    if (query) {
+      return person.name.toLowerCase().includes(query.toLowerCase())
+        || person.motherName?.toLowerCase().includes(query.toLowerCase())
+        || person.fatherName?.toLowerCase().includes(query.toLowerCase());
+    }
+
+    return true;
+  });
+
+  const filteredByCentury = filteredByQuery.filter(person => {
+    if (selectedCenturies.length > 0) {
+      const century = Math.ceil(person.died / 100);
+
+      return selectedCenturies.includes(century.toString());
+    }
+
+    return true;
+  });
+
+  const filteredPeople = filteredByCentury.filter(person => {
+    if (selectedSex) {
+      return person.sex === selectedSex;
+    }
+
+    return true;
+  });
+
+  const handleSort = (field: string) => {
+    let newOrder = 'asc';
+    const currentField = searchParams.get('sort');
+    const currentOrder = searchParams.get('order');
+
+    if (currentField === field) {
+      if (currentOrder === 'asc') {
+        newOrder = 'desc';
+      } else if (currentOrder === 'desc') {
+        searchParams.delete('sort');
+        searchParams.delete('order');
+        navigate(`/people?${searchParams.toString()}`);
+
+        return;
+      }
+    }
+
+    searchParams.set('sort', field);
+    searchParams.set('order', newOrder);
+    navigate(`/people?${searchParams.toString()}`);
+  };
+
+  useEffect(() => {
+    const newPeople = [...filteredPeople];
+    const sortField = searchParams.get('sort');
+    const sortOrder = searchParams.get('order');
+
+    if (sortField && sortOrder) {
+      const firstPerson = newPeople[0];
+      const sortFieldType = typeof firstPerson?.[sortField as keyof Person];
+
+      newPeople.sort((a, b) => {
+        const aValue = a[sortField as keyof Person];
+        const bValue = b[sortField as keyof Person];
+
+        if (sortFieldType === 'string') {
+          return (aValue as string).localeCompare(bValue as string);
+        }
+
+        if (sortFieldType === 'number') {
+          return (aValue as number) - (bValue as number);
+        }
+
+        return 0;
+      });
+
+      if (sortOrder === 'desc') {
+        newPeople.reverse();
+      }
+    }
+
+    setSortedPeople(newPeople);
+  }, [filteredPeople, searchParams]);
```

Comment:
are you sure you need that huge `useEffect`? do you even need an effect here?

Reply:
fair point 👌